spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/MappingSpannerConverter.java

```diff
@@ -31,78 +31,62 @@
-			.addAll(SpannerConverters.DEFAULT_SPANNER_READ_CONVERTERS).build();
+	private final ConverterAwareMappingSpannerEntityReader entityReader;
 
-	private static Set<Class> SPANNER_KEY_COMPATIBLE_TYPES = ImmutableSet
-			.<Class>builder().add(Boolean.class).add(Integer.class).add(Long.class)
-			.add(Float.class).add(Double.class).add(String.class).add(ByteArray.class)
-			.add(Timestamp.class).add(com.google.cloud.Date.class).build();
+	private final MappingSpannerWriteConverter entityWriter;
 
-	private final MappingSpannerReadConverter readConverter;
+	private final SpannerReadConverter readConverter;
 
-	private final MappingSpannerWriteConverter writeConverter;
+	private final SpannerWriteConverter writeConverter;
 
 	public MappingSpannerConverter(SpannerMappingContext spannerMappingContext) {
-		super(getCustomConversions(DEFAULT_SPANNER_CONVERTERS), null);
-		Assert.notNull(spannerMappingContext,
-				"A valid mapping context for Spanner is required.");
-		this.readConverter = new MappingSpannerReadConverter(spannerMappingContext,
-				getCustomConversions(SpannerConverters.DEFAULT_SPANNER_READ_CONVERTERS));
-		this.writeConverter = new MappingSpannerWriteConverter(spannerMappingContext,
-				getCustomConversions(SpannerConverters.DEFAULT_SPANNER_WRITE_CONVERTERS));
+		this(spannerMappingContext, null, null);
 	}
 
 	public MappingSpannerConverter(SpannerMappingContext spannerMappingContext,
 			Collection<Converter> writeConverters, Collection<Converter> readConverters) {
-		super(getCustomConversions(
-				ImmutableList.<Converter>builder().addAll(DEFAULT_SPANNER_CONVERTERS)
-						.addAll(readConverters).addAll(writeConverters).build()),
-				null);
 		Assert.notNull(spannerMappingContext,
 				"A valid mapping context for Spanner is required.");
-		this.readConverter = new MappingSpannerReadConverter(spannerMappingContext,
-				getCustomConversions(ImmutableList.<Converter>builder()
-						.addAll(SpannerConverters.DEFAULT_SPANNER_READ_CONVERTERS)
-						.addAll(readConverters).build()));
-		this.writeConverter = new MappingSpannerWriteConverter(spannerMappingContext,
-				getCustomConversions(ImmutableList.<Converter>builder()
-						.addAll(SpannerConverters.DEFAULT_SPANNER_WRITE_CONVERTERS)
-						.addAll(writeConverters).build()));
-	}
 
-	private static CustomConversions getCustomConversions(
-			Collection<Converter> converters) {
-		return new CustomConversions(StoreConversions.NONE, converters);
+		this.readConverter = new SpannerReadConverter(readConverters);
+		this.entityReader = new ConverterAwareMappingSpannerEntityReader(spannerMappingContext, this.readConverter);
+		this.writeConverter = new SpannerWriteConverter(writeConverters);
+		this.entityWriter = new MappingSpannerWriteConverter(spannerMappingContext, this.writeConverter);
 	}
 
 	@VisibleForTesting
-	MappingSpannerReadConverter getReadConverter() {
-		return this.readConverter;
+	ConverterAwareMappingSpannerEntityReader getEntityReader() {
```

Comment:
It looks like this method (and the one below it) aren't used anymore (including tests).